src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs

```diff
@@ -869,9 +869,11 @@ public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccoun
             string homeTenantId = account?.HomeAccountId?.TenantId;
             if (!string.IsNullOrEmpty(homeTenantId))
             {
+                // If it's an AAD account, only the AAD plugin should remove it
+                // If it's an MSA account - MSA plugin should remove it, but in MSA-PT scenarios it's still the AAD plugin
                 bool isMsaRequest = await IsMsaRequestAsync(
                    appConfig.Authority,
-                   appConfig.TenantId,
```

Comment:
So this was using the tenant id of the app instead of the tenant id of the home account so it caused an issue for MSA users?

Reply:
Yes, I think so. The original bug does not have clear repro steps, so I could not reproduce. But I think this may be a factor.